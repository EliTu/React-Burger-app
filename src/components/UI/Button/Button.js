import React from 'react';
import styles from './Button.module.css';
import PropTypes from 'prop-types';

const Button = ({ children, handleClick, type, toolBarButton }) => {
	// CSS Modules styles:
	const { Button } = styles;

	return (
		<button
			onClick={handleClick}
			className={[Button, styles[type], styles[toolBarButton]].join(' ')}
		>
			{children}
		</button>
	);
};

Button.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.arrayOf(PropTypes.node),
		PropTypes.node,
	]).isRequired,
	handleClick: PropTypes.func.isRequired,
	type: PropTypes.string,
	toolBarButton: PropTypes.string,
};

export default Button;
