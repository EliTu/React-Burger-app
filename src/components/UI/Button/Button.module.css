@import '../../../variables.css';

.Button {
    width: 95%;
    font-size: 1.15rem;
    align-self: baseline;
    margin: 0.5rem auto;
    padding: 0.5rem;
    border: 1px solid var(--coloring-opaque-dark);
    border-radius: 5px;
    background-color: initial;
    cursor: pointer;
    outline: none;
    transition: 0.1s all;
}

.Confirm {
    color: var(--glow-green-opaque-lighter);
    border: 2px solid var(--glow-green-opaque);
}

.Confirm:hover {
    color: var(--coloring-light);
    background-color: rgba(37, 168, 28, 0.781);
}

.Danger {
    color: var(--invalid-red-light);
    border: 2px solid var(--invalid-red-dark);
}

.Danger:hover {
    color: rgb(228, 25, 25);
    background-color: rgba(231, 159, 159, 0.74);
}

.Neutral {
    color: var(--coloring-opaque-dark);
    border: 2px solid var(--disabled-dark);
}

.Neutral:hover {
    color: rgb(228, 222, 222);
    background-color: rgba(175, 174, 174, 0.74)
}

/* Mobile - SideDrawer toggle */
@media (max-width: 500px) {
    .MenuToggle {
        color: var(--coloring-bright-light);
        font-weight: 800;
        font-size: 1.5rem;
        height: 100%;
        width: 20%;
        align-self: flex-end;
        margin: 0;
        padding: 0 0.9rem;
        border-bottom: none;
    }
}