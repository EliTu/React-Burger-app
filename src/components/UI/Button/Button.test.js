import React from 'react';
import Button from './Button';
import { shallow } from 'enzyme';

describe('<Button>', () => {
	let component = shallow(<Button children="abc" handleClick={() => []} />);

	it('should render the component without errors', () => {
		expect(component).toMatchSnapshot();
		expect(component.length).toBe(1);
		expect(component.length).not.toBe(2);
	});

	it('should render the children prop string without errors', () => {
		expect(component.children().text()).toBe('abc');
	});

	it('should set button styles according to the type prop', () => {
		component.setProps({ type: 'Confirm', toolBarButton: '' });

		expect(component).toMatchSnapshot();
		expect(component.hasClass('Button Confirm')).toBe(true);

		component.setProps({ type: 'Danger', toolBarButton: '' });
		expect(component).toMatchSnapshot();
		expect(component.hasClass('Button Danger')).toBe(true);
	});

	it('should set the toolBarButton style to MenuToggle on a small screen', () => {
		component.setProps({ type: '', toolBarButton: 'MenuToggle' });

		expect(component).toMatchSnapshot();
		expect(component.hasClass('Button  MenuToggle')).toBe(true);
	});
});
