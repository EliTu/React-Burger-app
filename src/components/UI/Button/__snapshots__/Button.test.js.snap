// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> should render the component without errors 1`] = `
<button
  className="Button undefined undefined"
  onClick={[Function]}
>
  abc
</button>
`;

exports[`<Button> should set button styles according to the type prop 1`] = `
<button
  className="Button Confirm "
  onClick={[Function]}
>
  abc
</button>
`;

exports[`<Button> should set button styles according to the type prop 2`] = `
<button
  className="Button Danger "
  onClick={[Function]}
>
  abc
</button>
`;

exports[`<Button> should set the toolBarButton style to MenuToggle on a small screen 1`] = `
<button
  className="Button  MenuToggle"
  onClick={[Function]}
>
  abc
</button>
`;
